File: backend/logsheet/service/LogsheetTypeService.java
package com.sunbeam.logsheet.service;

import com.sunbeam.logsheet.DTO.LogsheetTypeDTO;
import com.sunbeam.logsheet.DTO.ApiResponse;

import java.util.List;

public interface LogsheetTypeService {
    ApiResponse<LogsheetTypeDTO> addLogsheetType(LogsheetTypeDTO dto);
    ApiResponse<List<LogsheetTypeDTO>> getAllLogsheetTypes();
    ApiResponse<LogsheetTypeDTO> getLogsheetTypeById(Long id);
    ApiResponse<LogsheetTypeDTO> updateLogsheetType(Long id, LogsheetTypeDTO dto);
    ApiResponse<Boolean> deleteLogsheetType(Long id);
}

File: frontend/logsheet_fe/src/services/logsheetTypeService.js
import axios from "axios";
import { config } from "./config";

const API_URL = `${config.serverBaseUrl}/api/logsheet-types`;


const getAuthHeader = () => {
  const user = JSON.parse(sessionStorage.getItem("user")); 
  if (user && user.token) {
    return {
      Authorization: `Bearer ${user.token}`,
    };
  }
  return {};
};


export const getAllLogsheetTypes = async () => {
  return axios.get(API_URL,{ headers: getAuthHeader() });
};


export const addLogsheetType = async (logsheetType) => {
  return axios.post(API_URL, logsheetType,{ headers: getAuthHeader() });
};


export const updateLogsheetType = async (id, logsheetType) => {
  return axios.put(`${API_URL}/${id}`, logsheetType,{ headers: getAuthHeader() });
};


export const deleteLogsheetType = async (id) => {
  return axios.delete(`${API_URL}/${id}`,{ headers: getAuthHeader() });
};


export const getLogsheetTypeById = async (id) => {
  return axios.get(`${API_URL}/${id}`,{ headers: getAuthHeader() });
};
